lib/modules/datasource/rubygems/index.ts

```diff
@@ -57,39 +57,42 @@ export class RubyGemsDatasource extends Datasource {
         cachedVersions.type === 'not-supported' &&
         registryHostname !== 'rubygems.org'
       ) {
-        const hostname = registryHostname;
-        return hostname === 'rubygems.pkg.github.com' ||
-          hostname === 'gitlab.com'
-          ? await this.getDependencyFallback(registryUrl, packageName)
-          : await this.getDependency(registryUrl, packageName);
+        if (
+          registryHostname === 'rubygems.pkg.github.com' ||
+          registryHostname === 'gitlab.com'
+        ) {
+          return await this.getReleasesViaFallbackAPI(registryUrl, packageName);
+        }
+
+        const gemMetadata = await this.fetchGemMetadata(
+          registryUrl,
+          packageName
+        );
+        if (!gemMetadata) {
+          return await this.getReleasesViaFallbackAPI(registryUrl, packageName);
+        }
+
+        return await this.getReleasesViaAPI(
+          registryUrl,
+          packageName,
+          gemMetadata
+        );
       }
+
+      return null;
     } catch (error) {
       this.handleGenericErrors(error);
     }
-
-    return null;
   }
 
-  async getDependencyFallback(
+  async fetchGemMetadata(
```

Comment:
why function reordering, makes PR more complicated than it has to

Reply:
Sorry for that, but new ordering is much better

Reply:
ok, please do it in separate PR's next time.